src/components/TodoModal/TodoModal.tsx

```
@@ -2,41 +1,75 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 
 export const TodoModal: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<null | User>(null);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+  const closeCurrentTodo = () => {
+    dispatch(currentTodoActions.clearCurrentUser());
+  };
+
+  useEffect(() => {
+    setIsLoading(true);
+    if (currentTodo) {
+      (async () => {
+        try {
+          const result = await getUser(currentTodo.userId);
+
+          setUser(result);
+          setIsLoading(false);
+        } catch {}
```

Consider adding error handling in the `catch` block to handle potential errors from the `getUser` API call. This will help in providing feedback or logging errors instead of failing silently.